### DISCUSSION TITLE
Activating another micromamba environment upon entry

### DISCUSSION THREAD

**Discussion by calvinp0** [mamba-org/micromamba-docker#312] [Q&A]
Hey. So in our docker image with micromamba as the base image, we use the command 

```dockerfile
ARG MAMBA_DOCKERFILE_ACTIVATE=1
ENV ENV_NAME=rmg_env
```

And then after that, in our build we also install a couple more micromamba environments.

The docker builds perfectly but upon interaction it starts with the environment as rmg_env (which is to be expected). However, we want it so that upon entry of user, they will start off with one of the other installed environments. More specifically, `t3_env`.

Is it possible to do this?

**Comment by maresb** [comment]
Have you tried something like this?

```bash
docker run --rm -it -e ENV_NAME=te_env your-docker-image
```

**Comment by wholtz** [comment]
If you want an environment automatically activated on `docker run ...`, without passing `-e ENV_NAME=t3_env` as @maresb suggested, you'll have to name that environment `base`.

**Comment by npow** [comment]
I have similar use-case. I'm currently activating the default `base` environment via the entrypoint, but then the first thing it does is sets up another conda environment dynamically. How can I switch to that conda environment after it's setup? I keep getting `'micromamba' is running as a subprocess and can't modify the parent shell.` when I try to do `micromamba activate <ENV_NAME>`.

For example, I want to do something like the following in my entrypoint:

```
python trampoline.py <-- sets up <ENV_NAME>
micromamba activate <ENV_NAME>
python server.py
```

**Reply by wholtz** [reply]
Does this do what you need?

```
python trampoline.py <-- sets up <ENV_NAME>
micromamba run -n <ENV_NAME> python server.py
```

**Reply by maresb** [reply]
Alternatively,

```bash
source _activate_current_env.sh
python trampoline.py <-- sets up <ENV_NAME>
micromamba activate <ENV_NAME>
python server.py
```

The command `source _activate_current_env.sh` modifies your shell so that `micromamba` runs as a Bash function rather than the `/usr/local/bin/micromamba` executable. In order to be able to activate an environment in your current shell, `micromamba` needs to be running a Bash function.

In contrast, the `micromamba run` command is for one-off commands without shell activation.
